### ISSUE TITLE
UnrealIRCd 5.0.7 crashes on rehash when using m_autoban

### ISSUE THREAD

**Issue by Rixxan** [FuelRats/m_autoban#1]
## Description

When using m_autoban, and issuing the /rehash command (or, ./unrealircd rehash), the IRCd crashes and must be manually restarted. 
## Expected Behavior
UnrealIRCd's configuration file should be rehashed, and the server should remain online and operational for all users. 

## Actual Behavior
UnrealIRCd crashes, and all clients lose connection to the server. The server does not manually return to online status, and instead stays offline until manually restarted on the server.

## Steps to Reproduce

1. Install UnrealIRCd v. 5.0.7
2. Download and compile the latest commit of m_autoban.c (https://raw.githubusercontent.com/FuelRats/m_autoban/master/m_autoban.c)
3. MAKE and then COPY the new m_autoban.so to the IRCd's modules/third library (OF NOTE: There are some warnings during compile such as https://pastebin.com/DcvDaJGC. )
4. INSTALL the module in the UnrealIRCd server, and start the server. 
5. Issue the /rehash command (after verifying that m_autoban's functionality is in place)

## Context

Without this module, the server is able to work fine. With this module, it's impossible to make little, on-the-fly tweaks to Unrealircd's configuration without necessitating an entire server restart. 

## Your Environment

* Version used: MASTER branch as of commit f1bc4b0.
* Environment name and version (e.g. Chrome 39, node.js 5.4): UnrealIRCd-5.0.7
* Operating System and version (desktop or mobile): Ubuntu 20.04 LTS

**Comment by SuperManifolds** [comment]
Hi there! I am unable to reproduce this issue, our servers have rehashed many times whilst running the plugin, both using ./unrealircd rehash and using /rehash. Are you able to provide any kind of crash log from Unreal itself?

**Comment by Rixxan** [comment]
> 
> 
> Hi there! I am unable to reproduce this issue, our servers have rehashed many times whilst running the plugin, both using ./unrealircd rehash and using /rehash. Are you able to provide any kind of crash log from Unreal itself?

Hey there!

Here's the relevant section of the UnrealIRCd crash report, just replicated the issue with a fresh compile.

https://pastebin.com/rxvXFece

**Comment by modmenager** [comment]
module m_autoban ned update for unrealircd-5.0.7 

m_autoban.c: In function ‘getIPv6BanRange’:
m_autoban.c:205:29: warning: ‘__builtin___sprintf_chk’ may write a terminating nul past the end of the destination [-Wformat-overflow=]
         sprintf(output, "%s*", delimitedOutput);
                             ^
In file included from /usr/include/stdio.h:873,
                 from m_autoban.c:1:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:36:10: note: ‘__builtin___sprintf_chk’ output between 2 and 9 bytes into a destination of size 8
   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       __bos (__s), __fmt, __va_arg_pack ());
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
m_autoban.c: In function ‘getIPForNickname’:
m_autoban.c:246:19: warning: function returns an aggregate [-Waggregate-return]
 struct IPUserInfo getIPForNickname (char* nickname) {
                   ^~~~~~~~~~~~~~~~
m_autoban.c: In function ‘autoban_func’:
m_autoban.c:315:34: warning: function call has aggregate value [-Waggregate-return]
     struct IPUserInfo userInfo = getIPForNickname(banTarget);
                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~

**Comment by Rixxan** [comment]
Resolved in 5ff77bd.
